engineapi/proxy.go

```diff
@@ -81,9 +84,19 @@ func NewEngineClientProxy(im *longhorn.InstanceManager, logger logrus.FieldLogge
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
-	client, err := imclient.NewProxyClient(ctx, cancel, im.Status.IP, InstanceManagerProxyServiceDefaultPort)
+	client, err := imclient.NewProxyClientWithTLS(ctx, cancel, im.Status.IP, InstanceManagerProxyServiceDefaultPort,
+		filepath.Join(types.TLSDirectoryInContainer, types.TLSCAFile),
+		filepath.Join(types.TLSDirectoryInContainer, types.TLSCertFile),
+		filepath.Join(types.TLSDirectoryInContainer, types.TLSKeyFile),
+		"longhorn-backend.longhorn-system")
 	if err != nil {
-		return nil, err
+		logrus.WithError(err).Warnf("Falling back to non-tls client for Instance Manager Instance Service Client for %v, state: %v, IP: %v",
+			im.Name, im.Status.CurrentState, im.Status.IP)
+		client, err = imclient.NewProxyClient(ctx, cancel, im.Status.IP, InstanceManagerProxyServiceDefaultPort, nil)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to initialize Instance Manager Instance Service Client for %v, state: %v, IP: %v, TLS: %v",
```

Comment:
Proxy service client

Reply:
Is it necessary to have the state, IP, and TLS in the error? They've already been logged above. WDYT?

Reply:
Change the above log level to Trace, so I think recording the IP is still necessary.